### ISSUE TITLE
[SidePanel]: Update title to use TruncatedText (React and wc)

### ISSUE THREAD

**Issue by davidmenendez** [carbon-design-system/ibm-products#8233]
Currently `SidePanel` is using `title` to handle text overflow for long titles. It should be removed and the truncation should be handled with `TruncatedText`

Acceptance criteria

- [x] Remove all usage of `title` attribute from `SidePanel`
- [ ] Implement `TruncatedText` component to handle the title
- [x] Make necessary changes to styles and code to verify working implementation

**Comment by matthewgallo** [comment]
Wanted to add that after adding `workbreak` to the side panel `tltle`, I'm now seeing that this causes all words to break onto a new line even when you have a title that is not just a long string (like what was seen in the [original issue](https://github.com/carbon-design-system/ibm-products/issues/8201)). I believe using the truncated text component would fix this and we could remove `wordbreak`.

The only reason I noticed this was because of a change that got flagged by percy (maybe it still is good for something even if it flags changes we've already approved 😄)
